compiler/rustc_trait_selection/src/traits/select/candidate_assembly.rs

```diff
@@ -87,6 +87,12 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
             } else if lang_items.sized_trait() == Some(def_id) {
                 // Sized is never implementable by end-users, it is
                 // always automatically computed.
+
+                // Error type cannot possibly implement `Sized` (fixes #123154)
+                if let ty::Error(_) = obligation.predicate.skip_binder().self_ty().kind() {
+                    return Err(SelectionError::Unimplemented);
```

Comment:
Do we get better errors if we pick Overflow::ErrorReported?

Reply:
Didn't try. Let me have a look.

Reply:
From the looks of it it may reduce duplicates, but it is not preventing the ICE from happening. Will dig into it tomorrow.

Reply:
This has stopped the ICE eliminated duplicate diagnostics, but it has lead to https://github.com/rust-lang/rust/issues/68830 regressing. I'm investigating.